tests/test_import_status_endpoint.py

```diff
@@ -144,6 +152,68 @@ def should_retrieve_validation_failures(describe_step_function_mock: MagicMock)
         assert response == expected_response
 
 
+@patch("backend.step_function.STEP_FUNCTIONS_CLIENT.describe_execution")
+@patch("backend.step_function.S3CONTROL_CLIENT.describe_job")
+def should_report_s3_batch_upload_task_failures(
+    describe_s3_job_mock: MagicMock,
+    describe_step_function_mock: MagicMock,
+) -> None:
+    # Given
+    describe_step_function_mock.return_value = {
+        STATUS_KEY: JOB_STATUS_SUCCEEDED,
+        INPUT_KEY: dumps(
+            {DATASET_ID_KEY: any_dataset_id(), VERSION_ID_KEY: any_dataset_version_id()}
+        ),
+        OUTPUT_KEY: dumps(
+            {
+                VALIDATION_KEY: {SUCCESS_KEY: True},
+                IMPORT_DATASET_KEY: {
+                    METADATA_JOB_ID_KEY: any_job_id(),
+                    ASSET_JOB_ID_KEY: any_job_id(),
+                },
+            }
+        ),
+    }
+
+    describe_s3_job_mock.return_value = {
+        "Job": {
+            "Status": S3_BATCH_STATUS_COMPLETE,
+            "FailureReasons": [],
+            "ProgressSummary": {"NumberOfTasksFailed": 1},
+        }
+    }
+
+    expected_response = {
+        STATUS_CODE_KEY: HTTPStatus.OK,
+        BODY_KEY: {
+            STEP_FUNCTION_KEY: {STATUS_KEY: "Succeeded"},
+            VALIDATION_KEY: {STATUS_KEY: Outcome.PASSED.value, ERRORS_KEY: []},
+            METADATA_UPLOAD_KEY: {
+                STATUS_KEY: S3_BATCH_STATUS_FAILED,
+                ERRORS_KEY: {FAILED_TASKS_KEY: 1, FAILURE_REASONS_KEY: []},
+            },
+            ASSET_UPLOAD_KEY: {
+                STATUS_KEY: S3_BATCH_STATUS_FAILED,
+                ERRORS_KEY: {FAILED_TASKS_KEY: 1, FAILURE_REASONS_KEY: []},
```

Comment:
We should make sure the failed tasks are reported independently for assets and metadata files. You should be able to do this by creating a function which mimics [`describe_job`](https://boto3.amazonaws.com/v1/documentation/api/1.17.61/reference/services/s3control.html#S3Control.Client.describe_job) and returns two different values, something like this:

```python
def describe_job_mock(**kwargs: Any):
    if kwargs["JobId"] == metadata_job_id:
        return metadata_failed_task_count
    if kwargs["JobId"] == asset_job_id:
        return asset_failed_task_count
    assert False, f"Unknown job ID {kwargs['JobId']}"

describe_s3_job_mock.side_effect = describe_job_mock
```